File: capacitator/plugins/my-clarity-capacitator-plugin/ios/Sources/MyClarityCapacitatorPlugin/MyClarityCapacitatorPlugin.swift
import Foundation
import Capacitor

/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(MyClarityCapacitatorPlugin)
public class MyClarityCapacitatorPlugin: CAPPlugin, CAPBridgedPlugin {
    public let identifier = "MyClarityCapacitatorPlugin"
    public let jsName = "MyClarityCapacitator"
    public let pluginMethods: [CAPPluginMethod] = [
        CAPPluginMethod(name: "initialize", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "setCurrentScreenName", returnType: CAPPluginReturnPromise),
        CAPPluginMethod(name: "echo", returnType: CAPPluginReturnPromise)
    ]
    private var implementation: MyClarityCapacitator?

    override public func load() {
      self.implementation = MyClarityCapacitator(plugin: self)
    }
    
    @objc func initialize(_ call: CAPPluginCall) {
        let id = call.getString("id") ?? ""
        call.resolve([
            "id": implementation?.initialize(id) ?? ""
        ])
    }

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.resolve([
            "value": implementation?.echo(value) ?? ""
        ])
    }
    
    @objc func setCurrentScreenName(_ call: CAPPluginCall) {
        let id = call.getString("id") ?? ""
        call.resolve([
            "id": implementation?.setCurrentScreenName(id) ?? ""
        ])
    }

    // @objc private func onPause() {
    //     implementation?.handleOnPause()
    // }
    
    // @objc private func onResume() {
    //     implementation?.handleOnResume()
    // }
}

File: capacitator/plugins/my-clarity-capacitator-plugin/android/src/main/java/com/ecmo/bedside/plugins/clarity/MyClarityCapacitatorPlugin.java
package com.sweetrush.staging.elso.plugins.clarity;

import com.getcapacitor.JSObject;
import com.getcapacitor.Plugin;
import com.getcapacitor.PluginCall;
import com.getcapacitor.PluginMethod;
import com.getcapacitor.annotation.CapacitorPlugin;

@CapacitorPlugin(name = "MyClarityCapacitator")
public class MyClarityCapacitatorPlugin extends Plugin {

    private MyClarityCapacitator implementation;

    @Override
    public void load() {
        implementation = new MyClarityCapacitator(getContext());
    }

    @Override
    public void handleOnPause() {
        super.handleOnPause();
        implementation.handleOnPause();
    }

    @Override
    public void handleOnResume() {
        super.handleOnResume();
        implementation.handleOnResume();
    }

    @PluginMethod
    public void initialize(PluginCall call) {
        String id = call.getString("id");
        
        JSObject ret = new JSObject();
        ret.put("id", implementation.initialize(id));
        call.resolve(ret);
    }
    
    @PluginMethod
    public void setCurrentScreenName(PluginCall call) {
        String id = call.getString("id");
        
        JSObject ret = new JSObject();
        ret.put("id", implementation.setCurrentScreenName(id));
        call.resolve(ret);
    }

    @PluginMethod
    public void echo(PluginCall call) {
        String value = call.getString("value");

        JSObject ret = new JSObject();
        ret.put("value", implementation.echo(value));
        call.resolve(ret);
    }
}
